### PR TITLE
Use improved OpenAPI enums

### PR DESCRIPTION
**Pull request by 1lutz**

### CODE REVIEW

**Comment by 1lutz** [comment]
[OpenAPI Enums verbessern](https://trello.com/c/WmlSSWrF/1675-openapi-enums-verbessern)

File: geoengine/colorizer.py [hunk 1]
```diff
@@ -238,19 +244,19 @@ class LinearGradientColorizer(Colorizer):
     under_color: Rgba
 
     @staticmethod
-    def from_response_linear(response: geoengine_openapi_client.LinearGradientWithType) -> LinearGradientColorizer:
+    def from_response_linear(response: geoengine_openapi_client.LinearGradient) -> LinearGradientColorizer:
         """Create a colorizer from a response."""
         breakpoints = [ColorBreakpoint.from_response(breakpoint) for breakpoint in response.breakpoints]
         return LinearGradientColorizer(
-            no_data_color=response.no_data_color,
+            no_data_color=to_rgba(response.no_data_color),
```
Comment (michaelmattig) [inline geoengine/colorizer.py:251]: why do you need to convert the type now? Can we adjust the type in OpenAPI so that we get a Rgba directly without conversion?

File: geoengine/error.py [hunk 2]
```diff
@@ -19,10 +19,12 @@ class GeoEngineException(Exception):
     def __init__(self, response: Union[geoengine_openapi_client.ApiException, Dict[str, str]]) -> None:
         super().__init__()
 
-        if isinstance(response, geoengine_openapi_client.ApiException):
+        if isinstance(response, dict):
```
Comment (michaelmattig) [inline geoengine/error.py:22]: why did you need to change this? What happened to the ApiException type in the openapi client?

File: geoengine/raster.py [hunk 3]
```diff
@@ -191,9 +191,9 @@ def spatial_resolution(self) -> gety.SpatialResolution:
     def from_ge_record_batch(record_batch: pa.RecordBatch) -> RasterTile2D:
         '''Create a RasterTile2D from an Arrow record batch recieved from the Geo Engine'''
         metadata = record_batch.schema.metadata
-        geo_transform = gety.GeoTransform.from_response(
-            geoengine_openapi_client.GdalDatasetGeoTransform.from_json(metadata[b'geoTransform'])
-        )
+        inner = geoengine_openapi_client.GdalDatasetGeoTransform.from_json(metadata[b'geoTransform'])
+        assert inner is not None
```
Comment (michaelmattig) [inline geoengine/raster.py:195]: add error message

File: tests/util.py [hunk 4]
```diff
@@ -5,6 +5,18 @@
...
 
 
+def is_url_match(url1: str, url2: str) -> bool:
+    '''Checks if two urls point to the same resource'''
+    parsed1 = urllib3.util.parse_url(url1)
+    parsed2 = urllib3.util.parse_url(url2)
+    return (parsed1.host == parsed2.host
+            and parsed1.port == parsed2.port
+            and parsed1.path == parsed2.path
+            and (parsed1.query == parsed2.query
```
Comment (michaelmattig) [inline tests/util.py:15]: I think you can do

query1 = parse_qs(parsed1.query)
query2 = parse_qs(parsed2.query)
    
and then just compare the query objects:
query1 == query2

please add a test for the method though.

File: setup.cfg [hunk 5]
```diff
@@ -18,7 +18,7 @@ package_dir =
 packages = find:
 python_requires = >=3.8
 install_requires =
-    geoengine-openapi-client == 0.0.4
+    geoengine-openapi-client @ git+https://github.com/1lutz/openapi-client.git/#subdirectory=python # TODO update when merged
```
Comment (michaelmattig) [inline setup.cfg:21]: remember to change

File: geoengine/colorizer.py (hunk 1 above)
Comment (1lutz) [reply geoengine/colorizer.py:251]: The newer generator outputs a list with a limited length. This is different from Rgba, because that type is a tuple. Because of this a conversion is needed.

Because i reverted to the older generator, I was able to remove the type conversion.

File: geoengine/error.py (hunk 2 above)
Comment (1lutz) [reply geoengine/error.py:22]: I don't remember, but I was able to revert this.

File: setup.cfg [hunk 6]
```diff
@@ -45,7 +45,7 @@ dev =
     pylint >=2.11,<3 # code linter
     setuptools >=42,<68
     twine >=3.4,<5 # PyPI
-    types-pkg-resources >=0.1.3 # mypy type hints
+    types-setuptools >= 71.1.0.20240726 # mypy type hints
```
Comment (1lutz) [inline setup.cfg:48]: The other package was deprecated and I could not install it anymore. So I included the successor.

File: geoengine/colorizer.py (hunk 1 above)
Comment (1lutz) [reply geoengine/colorizer.py]: [9235965](https://github.com/geo-engine/geoengine-python/pull/187/commits/923596557d6bf89ccf3a42014535a797cf0ff1a4)

File: geoengine/error.py (hunk 2 above)
Comment (1lutz) [reply geoengine/error.py]: [9235965](https://github.com/geo-engine/geoengine-python/pull/187/commits/923596557d6bf89ccf3a42014535a797cf0ff1a4)

File: geoengine/raster.py (hunk 3 above)
Comment (1lutz) [reply geoengine/raster.py]: [9235965](https://github.com/geo-engine/geoengine-python/pull/187/commits/923596557d6bf89ccf3a42014535a797cf0ff1a4)

File: geoengine/types.py [hunk 7]
```diff
@@ -712,7 +712,7 @@ def from_response_raster(
             spatial_resolution = SpatialResolution.from_response(response.resolution)
 
         return RasterResultDescriptor(
-            data_type=data_type,
+            data_type=data_type,  # type: ignore
```
[the remark this answers is not in the document]
Comment (1lutz) [reply geoengine/types.py]: [9235965](https://github.com/geo-engine/geoengine-python/pull/187/commits/923596557d6bf89ccf3a42014535a797cf0ff1a4)

File: tests/util.py (hunk 4 above)
Comment (1lutz) [reply tests/util.py:16]: I don't know what function you mean. It is not available in urllib3. Maybe it is from the older urllib package but I won't include both versions.

[9235965](https://github.com/geo-engine/geoengine-python/pull/187/commits/923596557d6bf89ccf3a42014535a797cf0ff1a4)

Comment (michaelmattig) [reply tests/util.py:16]: I mean this function: from urllib.parse import parse_qs
and even in the urllib3 docs they use urllib.parse alongside urllib3 https://urllib3.readthedocs.io/en/stable/user-guide.html#query-parameters

Comment (1lutz) [reply tests/util.py]: [2cd6df4](https://github.com/geo-engine/geoengine-python/pull/187/commits/2cd6df407b536b20a5fb6458ea144710a60bab6a)

File: setup.cfg (hunk 5 above)
Comment (1lutz) [reply setup.cfg]: 651f97b0f2113c32e9636ff72423f779f31ea32b

**Review by michaelmattig — APPROVED** [review]
